src/Tribe/Tickets_View.php

```diff
@@ -1456,23 +1459,69 @@ public function get_my_tickets_link_data( int $event_id, int $user_id ): array {
 	}
 
 	/**
-	 * Prevent canonical redirect from stripping tribe-edit-orders parameter.
-	 *
-	 * When WordPress performs canonical redirects for pages accessed via ?p=ID,
-	 * it strips custom query parameters like tribe-edit-orders. This method
-	 * preserves the tribe-edit-orders parameter during such redirects.
+	 * Preserve pretty URL format in canonical redirect for tickets pages.
 	 *
 	 * @since TBD
 	 *
 	 * @param string $redirect_url The URL to redirect to.
 	 * @return string The URL to redirect to.
 	 */
 	public function preserve_tickets_parameter_in_canonical_redirect( $redirect_url ) {
-		// If we have the tribe-edit-orders parameter, preserve it in the redirect.
+		// If we have the tribe-edit-orders parameter, redirect to pretty URL format.
 		if ( ! empty( $redirect_url ) && get_query_var( 'tribe-edit-orders' ) ) {
+			$post_id = get_query_var( 'p' );
+			
+			if ( $post_id ) {
+				// Check if we have pretty permalinks enabled.
+				$has_plain_permalink = '' === get_option( 'permalink_structure' );
+				
+				if ( ! $has_plain_permalink ) {
+					// Redirect to the pretty URL format.
+					return home_url( '/tickets/' . $post_id );
+				}
+			}
+			
+			// Fallback: preserve the parameter in query string format.
 			$redirect_url = add_query_arg( 'tribe-edit-orders', 1, $redirect_url );
 		}
 
 		return $redirect_url;
 	}
+
+	/**
+	 * Handle tickets page requests to prevent canonical redirect issues.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $query_vars The query variables.
+	 * @return array The modified query variables.
+	 */
+	public function handle_tickets_request( $query_vars ) {
+		// Bail early if this is not a tickets page request.
+		if ( ! isset( $query_vars['tribe-edit-orders'] ) || ! $query_vars['tribe-edit-orders'] ) {
```

Comment:
This is an `empty` equivalent. What the empty is doing is this:

```php
! empty( $array['offset'] ) === isset( $array['offset'] ) && $array['offset']
```

so you can simply do:
```suggestion
		if ( empty( $query_vars['tribe-edit-orders'] ) ) {
```